src/library/scala/util/control/Exception.scala

```diff
@@ -13,24 +13,125 @@ package control
+ *
+ *  val goodUrl = "http://www.scala-lang.org/"
+ *  // Prints page content
+ *  withThrowableLogging { printUrl(goodUrl) }
+ *  }}}
+ *
+ *  Use `unwrapping` to create a `Catch` that unwraps exceptions before rethrowing.
+ *  {{{
+ *  class AppException(cause: Throwable) extends RuntimeException(cause)
+ *
+ *  val unwrappingCatch = unwrapping(classOf[AppException])
+ *
+ *  def calcResult: Int = throw new AppException(new NullPointerException)
+ *
+ *  // Throws NPE not AppException
+ *  val result = unwrappingCatch(calcResult)
+ *  }}}
+ *
+ *  Use `failAsValue` to provide a default when a specified exception is caught.
+ *
+ *  {{{
+ *  val inputDefaulting = failAsValue(classOf[NumberFormatException])(0)
+ *  val candidatePick = "seven" // scala.io.StdIn.readLine()
+ *
+ *  val pick = inputDefaulting(candidatePick.toInt) // 0
  *  }}}
  *
- *  This class differs from `scala.util.Try` in that it focuses on composing exception handlers rather than
- *  composing behavior.   All behavior should be composed first and fed to a `Catch` object using one of the
- *  `opt` or `either` methods.
+ *  Compose multiple `Catch`s with `or` to build a `Catch` that provides default values varied by exception.
+ *  {{{
+ *  val formatDefaulting = failAsValue(classOf[NumberFormatException])(0)
+ *  val nullDefaulting = failAsValue(classOf[NullPointerException])(-1)
+ *  val otherDefaulting = nonFatalCatch withApply(_ => -100)
+ *
+ *  val combinedDefaulting = formatDefaulting or nullDefaulting or otherDefaulting
+ *
+ *  def p(s: String): Int = s.length * s.toInt
+ *
+ *  combinedDefaulting(p("tenty-nine")) // 0
+ *
+ *  combinedDefaulting(p(null: String)) // -1
+ *
+ *  combinedDefaulting(throw new IllegalStateException) // -100
+ *
+ *  combinedDefaulting(p("11")) // 22
+ *  }}}
+ *
+ *  @groupname composition-catch Catch behaviour composition
+ *  @groupprio composition-catch 10
+ *  @groupdesc composition-catch Build Catch objects from exception lists and catch logic
+ *
+ *  @groupname composition-finally Finally behaviour composition
+ *  @groupprio composition-finally 20
+ *  @groupdesc composition-finally Build Catch objects from finally logic
+ *
+ *  @groupname canned-behaviour General purpose catch objects
+ *  @groupprio canned-behaviour 30
+ *  @groupdesc canned-behaviour Catch objects with predefined behaviour. Use combinator methods to compose additional behaviour.
+ *
+ *  @groupname dsl DSL behaviour composition
+ *  @groupprio dsl 40
+ *  @groupdesc dsl Expressive Catch behaviour composition
+ *
+ *  @groupname composition-catch-promiscuously Promiscuous Catch behaviours
+ *  @groupprio composition-catch-promiscuously 50
+ *  @groupdesc composition-catch-promiscuously Useful if catching `ControlThrowable` or `InterruptedException` is required.
+ *
+ *  @groupname logic-container Logic Containers
+ *  @groupprio logic-container 60
+ *  @groupdesc logic-container Containers for catch and finally behaviour.
  *
- *  @author Paul Phillips
```

Comment:
I don't normally like author tags, but I think this one is justified. This stuff is especially paulpy. It's up to the build not emit the doc tags. And other major contributors do add their names at times, I guess for self-branding purposes.